dataflow_lite/include/dataflow_lite/cloudwatch/cloudwatch_service.h

```diff
@@ -215,6 +218,14 @@ virtual T convertInputToBatched(const D &input) = 0;
  * Main workhorse thread that dequeues from the source and calls Task run.
  */
 void work() {
```

Comment:
Why not add the delay to the publisher? Since this is specific to CloudWatch logs why not implement in `LogPublisher::publishData` as a precondition to publish?